staticfile.go

```diff
@@ -93,33 +94,47 @@ func serverStaticRouter(ctx *context.Context) {
 }
 
 type serveContentHolder struct {
-	data     []byte
-	modTime  time.Time
-	size     int64
-	encoding string
+	data       []byte
+	modTime    time.Time
+	size       int64
+	originSize int64 //original file size:to judge file changed
+	encoding   string
 }
 
 type serveContentReader struct {
 	*bytes.Reader
 }
 
+const (
+	//max file size to cache,default: 100k
+	MaxCacheFileSize int = 1024 * 100
+	//max file count to cache,default: 1000
+	MaxCacheFileCount int = 1000
+)
+
 var (
-	staticFileMap = make(map[string]*serveContentHolder)
-	mapLock       sync.RWMutex
+	staticFileLruCache, _ = lru.New(MaxCacheFileCount)
+	lruLock               sync.RWMutex
 )
 
 func openFile(filePath string, fi os.FileInfo, acceptEncoding string) (bool, string, *serveContentHolder, *serveContentReader, error) {
```

Comment:
In order to avoid OOM, I think we should not cache big file, for example, the size > 1M should not be cached. `1M` could be configurable.

Reply:
Limitation in here.
In order to limit file size and file numbers
![image](https://user-images.githubusercontent.com/6133860/83492155-6f4ced80-a4e5-11ea-84ae-2c45f200623c.png)

Reply:
> In order to avoid OOM, I think we should not cache big file, for example, the size > 1M should not be cached. `1M` could be configurable.

Already push some commits about making static file cache size configurable